analyzers/src/SonarAnalyzer.CSharp/Rules/UseValueParameter.cs

```diff
@@ -82,37 +78,22 @@ private static bool IsAccessorValue(IdentifierNameSyntax identifier, SemanticMod
                 return false;
             }
 
-            return semanticModel.GetSymbolInfo(identifier).Symbol is IParameterSymbol parameter &&
-                parameter.IsImplicitlyDeclared;
+            return semanticModel.GetSymbolInfo(identifier).Symbol is IParameterSymbol { IsImplicitlyDeclared: true };
         }
 
         private static string GetAccessorType(AccessorDeclarationSyntax accessorDeclaration)
         {
-            if (accessorDeclaration.IsKind(SyntaxKind.AddAccessorDeclaration) ||
-                accessorDeclaration.IsKind(SyntaxKind.RemoveAccessorDeclaration))
+            if (accessorDeclaration.IsAnyKind(SyntaxKind.AddAccessorDeclaration, SyntaxKind.RemoveAccessorDeclaration))
             {
                 return "event";
             }
 
-            var accessorList = accessorDeclaration.Parent;
-            if (accessorList == null)
-            {
-                return null;
-            }
-
-            var indexerOrProperty = accessorList.Parent;
-            if (indexerOrProperty is IndexerDeclarationSyntax)
+            return accessorDeclaration.Parent?.Parent switch
             {
-                return "indexer set";
-            }
-            else if (indexerOrProperty is PropertyDeclarationSyntax)
-            {
-                return "property set";
-            }
-            else
-            {
-                return null;
-            }
+                IndexerDeclarationSyntax _ => "indexer set",
+                PropertyDeclarationSyntax _ => "property set",
```

Comment:
Shouldn't there be a message for `property init`? Let's add `{{Error message}}` in UTs to see how it will look like.